### PR TITLE
Settings - Blockchain portfolio

### PR DESCRIPTION
**Pull request by NovacAndrei**

### CODE REVIEW

File: src/redux/app/actions.ts [hunk 1]
```diff
@@ -25,9 +29,23 @@ export const toggleTestNet = () => {
...
 
-export const setNetworkTestNetChainId = (blockchain: string, chainId: number) => {
+export const setNetworkTestNetChainId = (blockchain: Blockchain, chainId: number) => {
     return {
         type: APP_SET_NETWORK_TEST_NET_CHAIN_ID,
         data: { blockchain, chainId }
     };
 };
+
+export const toggleNetwork = (blockchain: Blockchain) => {
```
Comment (krisboit) [inline src/redux/app/actions.ts]: `toogleBlockchainActive`

File: src/redux/app/actions.ts [hunk 2]
```diff
@@ -25,9 +29,23 @@ export const toggleTestNet = () => {
...
+    return {
+        type: APP_TOGGLE_NETWORK,
+        data: { blockchain }
+    };
+};
+
+export const sortNetworks = (sortedNetworks: Array<{ blockchain: Blockchain; order: number }>) => {
```
Comment (krisboit) [inline src/redux/app/actions.ts]: `updateBlockchianOrder`

File: src/redux/app/actions.ts [hunk 3]
```diff
@@ -1,8 +1,12 @@
+import { Blockchain } from '../../core/blockchain/types';
+
 // actions consts
 export const APP_SWITCH_WALLET = 'APP_STATE_SWITCH_WALLET';
 export const APP_SET_TOS_VERSION = 'APP_SET_TOS_VERSION';
 export const APP_SET_TEST_NET = 'APP_SET_TEST_NET';
 export const APP_SET_NETWORK_TEST_NET_CHAIN_ID = 'APP_SET_NETWORK_TEST_NET_CHAIN_ID';
+export const APP_TOGGLE_NETWORK = 'APP_TOGGLE_NETWORK';
```
Comment (krisboit) [inline src/redux/app/actions.ts]: `APP_TOGGLE_BLOCKCHAIN`

File: src/redux/app/actions.ts [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
+export const APP_SORT_NETWORKS = 'APP_SORT_NETWORKS';
```
Comment (krisboit) [inline src/redux/app/actions.ts]: `APP_UPDATE_BLOCKCHAIN_ORDER`

File: src/redux/app/actions.ts (hunk 2 above)
Comment (krisboit) [reply src/redux/app/actions.ts]: i think you can use `Array<Blockchain>`, as it is now i expect values to be:
```ts
[
    {blockchain: 'ZILLIQA', order: 0},
    {blockchain: 'ETHEREUM', order: 1}
]
```
`order` is the same with element index

File: src/redux/app/reducer.ts [hunk 5]
```diff
@@ -15,10 +17,14 @@ const intialState: IAppState = {
     testNet: true,
     networks: {
         [Blockchain.ETHEREUM]: {
```
Comment (krisboit) [inline src/redux/app/reducer.ts]: we have to decide if the ordering affects all wallets or we have ordering for each wallet separatly

Comment (krisboit) [reply src/redux/app/reducer.ts]: also we should add a new property `blockchains` on the same level with `networks`. `networks` should have only networks info.

File: src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        backgroundColor: themes[theme].colors.header
+    }
+});
+
+export class BlockchainPortfolioComponent extends React.Component<
+    INavigationProps & IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>
+> {
+    public static navigationOptions = navigationOptions;
+
+    public renderNetwork = ({ item, index, move, moveEnd, isActive }: any) => {
```
Comment (krisboit) [inline src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx]: please don't use arrow functions in classes
we definitely need tslint rule :)

File: src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        return (
+            <View style={styles.container}>
+                <DraggableFlatList
+                    data={this.props.networks}
+                    renderItem={this.renderNetwork}
+                    keyExtractor={(item: any) => `${item.value.order}`}
+                    scrollPercent={5}
+                    onMoveEnd={({ data }) =>
+                        this.props.sortNetworks(
+                            data.reduce((array, blockchain, index) => {
```
Comment (krisboit) [inline src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx]: i think you can use `map()` here

File: src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+const mapDispatchToProps = {
+    toggleNetwork,
+    sortNetworks
+};
+
+const mapStateToProps = (state: IReduxState) => {
+    const networks = Object.keys(state.app.networks)
+        .reduce((array, key) => {
```
Comment (krisboit) [inline src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx]: use `map()` instead

File: src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    sortNetworks: typeof sortNetworks;
+}
+
+const mapDispatchToProps = {
+    toggleNetwork,
+    sortNetworks
+};
+
+const mapStateToProps = (state: IReduxState) => {
+    const networks = Object.keys(state.app.networks)
```
Comment (krisboit) [inline src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx]: `networks` type is different from the type of `networks` prop.

File: src/components/splash-screen/styles.ts [hunk 10]
```diff
@@ -15,6 +15,6 @@ export default (theme: ITheme) =>
         lottie: {
             ...StyleSheet.absoluteFillObject,
             height: '100%',
-            top: -150
+            top: -BASE_DIMENSION * 10
```
Comment (maricalucian) [inline src/components/splash-screen/styles.ts]: This dimension needs to be exact? Is needed so the animation exists screen to the top?
If not, is better to specify larger dimension in percents.

File: src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx [hunk 11]
```diff
@@ -9,31 +9,27 @@ import { Icon } from '../../../components/icon';
...
-    const networks = Object.keys(state.app.networks)
-        .reduce((array, key) => {
-            return [...array, { key, value: state.app.networks[key] }];
-        }, [])
-        .sort((a, b) => a.value.order - b.value.order);
-
     return {
-        networks
+        blockchains: Object.keys(state.app.blockchains)
+            .reduce((array, key) => [...array, { key, value: state.app.blockchains[key] }], [])
```
[the remark this answers is not in the document]
Comment (NovacAndrei) [reply src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx]: Changed to `map`, easier to understand this way

File: .github/workflows/main.yml [hunk 12]
```diff
@@ -46,8 +46,6 @@ jobs:
       - uses: actions/setup-ruby@v1
         with:
           ruby-version: "2.x"
-      - name: Generate firebase config file
```
Comment (krisboit) [inline .github/workflows/main.yml]: this shouldn't be removed

File: .github/workflows/main.yml [hunk 13]
```diff
@@ -88,8 +86,8 @@ jobs:
           ssh-private-key: ${{ secrets.SSH_CERTS }}
       - name: Add to host
         run: ssh-keyscan gitlab.com >> ~/.ssh/known_hosts
-      - name: Generate firebase config file
```
Comment (krisboit) [inline .github/workflows/main.yml]: same here

File: .github/workflows/main.yml [hunk 14]
```diff
@@ -88,8 +86,8 @@ jobs:
...
-        run: echo '${{ secrets.FB_CONFIG_IOS_BETA }}' > ios/GoogleService-Info.plist 
+      - name: Checkout
```
Comment (krisboit) [inline .github/workflows/main.yml]: this shouldn't be added back

**Pull request closed without merging**